packages/table/src/table-column/index.ts

```diff
@@ -209,20 +210,15 @@ export default defineComponent({
     instance.columnConfig = columnConfig
     return
   },
-  render() {
+  render () {
+    let children
     try {
-      return h(
-        'div',
-        this.$slots.default?.({
-          store: {},
-          _self: {},
-          column: {},
-          row: {},
-          $index: -1,
-        }),
-      )
+      children = this.$slots
+        .default?.()
+        .filter(childNode => childNode.type?.name === 'ElTableColumn')
     } catch {
```

Comment:
You need to handle `v-for` when the root tag is `Fragment`, you have to filter at least 2 twice.

Reply:
For example, when use give you template like:
```vue
<el-table>
  <el-column v-for="" />
</el-table>
```

Reply:
The expression `this.$slots.default?.()` will resolve a value: `VNode<Fragment>` which only have length of 1 and its type will be `Fragment === childNode.type.name`

Reply:
yes, now it's fixed